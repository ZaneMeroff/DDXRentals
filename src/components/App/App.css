* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}

/* login page */
.app-home {
  display: flex;
  background-image: url('../../data/DDR-Wallpaper.jpg');
  background-size: cover;
  background-position: bottom;
  height: 100vh;
  text-align: center;
  width: 100%;
}

.app-all {
  overflow: hidden;
}

#app-landing {
  overflow: hidden;
}

#app-map {
  background-color: #202020;
  background-size: cover;
}

#app-listing {
  overflow-y: scroll;
}

#app-account {
  overflow: hidden;
}
