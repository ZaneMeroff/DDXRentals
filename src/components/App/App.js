import React, { Component } from 'react';
import Form from '../Form/Form.js';
import SmallListingCard from '../SmallListingCard/SmallListingCard.js';
import Animation from '../Animation/Animation.js';
import Map from '../Map/Map.js';
import Nav from '../Nav/Nav.js';
import Neighborhood from '../Neighborhood/Neighborhood.js';
import { fetchRentalAreaData } from '../../helpers.js';
import './App.css';

class App extends Component {
  constructor() {
    super();
    this.state = {
      areaNames: {},
      areaDetails: {},
      listings: {},
      chosenHood: {
        id: 0,
        name: 'Select An Area on the Left to View Full Details',
        location: null,
        about: '',
        listings: []
      },
      hoods: {}
    }
  }

  componentDidMount() {
  fetch('http://localhost:3001/api/v1/areas')
    .then(response => response.json())
    .then(areaNamesData => {
      this.setState({areaNames: areaNamesData })
      this.updateListings();
      this.updateAreaDetails(areaNamesData.areas)
    })
    .catch(error => window.alert(`There was an error: ${error}`))
  }

  updateListings() {
    fetch('http://localhost:3001/api/v1/listings')
      .then(response => response.json())
      .then(listingInfo => this.setState( {listings: {listingInfo}} ))
      .catch(error => window.alert(`There was an error: ${error}`))
  }

  updateAreaDetails(areaDetails) {
    const selectedArea = fetchRentalAreaData(areaDetails)
    Promise.all(selectedArea)
      .then(data => {
        this.setState({ areaDetails: data }, () => this.addAreaNicknameToHoodz())
        this.addHoodData();
      })
  }

  addHoodData() {
    this.state.areaDetails.forEach(area => {
      this.setState({ hoods: { ...this.state.hoods, [area.name]: area } })
    });
  }


  addAreaNicknameToHoodz() {
    let details = this.state.areaNames.areas.reduce((acc, area) => {
      let id = area.details.split('/');
      let areaObj = {nickName: area.area, id: id[4]};
      this.state.areaDetails.forEach( zone => {
      if (parseInt(areaObj.id) === zone.id) {
          acc.push({...zone, nickName: areaObj.nickName});
        }
      })
      return acc;
    }, []);
    this.setState({ areaDetails: details });
  }

  updateNeighborhoodInfo = (zoneString) => {
    console.log(this.state.areaDetails);
    this.setState({ chosenHood: this.state.hoods[zoneString] });
  }

  render () {
    return (
      <div className="app">
        <Nav />
        <Map updateNeighborhoodInfo={this.updateNeighborhoodInfo} />
        <Neighborhood areas={this.state.chosenHood} />
        <SmallListingCard />

      </div>
    );
  }
}

export default App;
