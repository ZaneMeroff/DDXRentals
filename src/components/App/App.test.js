import React from 'react';
import ReactDOM from 'react-dom';
import { MemoryRouter } from 'react-router';
import App from './App';
import { shallow, mount } from 'enzyme';

describe('App', () => {
  let wrapper, mockArea;
  let memoryRouter;

  beforeEach(() => {
    wrapper = shallow(<App />)
    memoryRouter = mount(
    <MemoryRouter initialEntries={[ '/' ]}>
      <App/>
    </MemoryRouter>
  )
  mockArea =  [{
    listing_id: 3, area_id: 590, name: 'Hip RiNo Party Spot',
    address: {
      street: '2250 Lawrence St',
      zip: '80205'
    },
    details: {
      neighborhood_id: 5124122, superhost: true, seller_source: '91jss1', beds: 3, baths: 2.5, cost_per_night: 420, features: [
        'hot tub',
        'espresso machine'
      ],
    },
    dev_id: 'u4gh2j', area: 'rino', db_connect: 834470
  }]
})


  it('should be an instance of App component', () => {
      expect(wrapper).toMatchSnapshot();
  });

  it('should update its areaDetails state', () => {
    wrapper.instance().updateAreaDetails(mockArea);
    expect(wrapper.state('areaDetails')).toEqual(mockArea)
  });

  it('should update state when saveHoodData is invoked', () => {
    wrapper.instance().addHoodData = jest.fn();
    wrapper.instance().addHoodData();
    expect(wrapper.instance().addHoodData).toHaveBeenCalled();
  });

  it('should saveUserData', () => {
    const mockUserInfo = { email: 'icantsee@gmail.com', name: 'Stevie Wonder', value: 'Business' }
    wrapper.instance().saveUserData(mockUserInfo)
    expect(wrapper.state('userData')).toEqual(mockUserInfo)
  });

  it('should update state when addNicknamesToHoodz is invoked', () => {
    wrapper.instance().addAreaNicknameToHoodz = jest.fn();
    wrapper.instance().addAreaNicknameToHoodz();
    expect(wrapper.instance().addAreaNicknameToHoodz).toHaveBeenCalled();
  });

  it('should update state when updateNeighborhoodInfo is invoked', () => {
    const mockZone = 'River North';
    wrapper.instance().updateNeighborhoodInfo = jest.fn();
    wrapper.instance().updateNeighborhoodInfo(mockZone);
    expect(wrapper.instance().updateNeighborhoodInfo).toHaveBeenCalled();
  });

  it('should update state when addFavorite is invoked', () => {
    wrapper.instance().addFavorite = jest.fn();
    wrapper.instance().addFavorite(mockArea);
    expect(wrapper.instance().addFavorite).toHaveBeenCalled();
  })

});
