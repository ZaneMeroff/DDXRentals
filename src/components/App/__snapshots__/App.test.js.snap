// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should be an instance of App component 1`] = `
<main>
  <div>
    <Route
      exact={true}
      path="/"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/"
      render={[Function]}
    />
  </div>
  <Route
    path="/"
    render={[Function]}
  />
  <div
    className="app-map"
  >
    <Route
      exact={true}
      path="/map"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/map"
      render={[Function]}
    />
  </div>
  <div>
    <Route
      exact={true}
      path="/listings"
      render={[Function]}
    />
  </div>
  <div>
    <Route
      exact={true}
      path="/account"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/account"
      render={[Function]}
    />
  </div>
</main>
`;
